File: t10.cpp
// Write your code here
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main()
{
  int num = 0;
  cin >> num;
  for(;num>0;num--){
    string s;
    cin >> s;
    int len = s.length();
    bool isPal = true;//false;
    
    for(int i=0;i<len;i++){
      int j = len-1-i;
      if(i>=j)
        break;
      if(s[i]!=s[j]){
        isPal = false;
        break;
      }
    }
    if(isPal){
      cout<<"YES ";
      if(len%2 == 0)
        cout<<"EVEN";
      else
        cout<<"ODD";
    }
    else
      cout<<"NO";
    cout<<"\n";
  }
}

File: t10.py
if __name__ == '__main__':
  num = input()
  for n in range(int(num)):
    s = input()
    # print (s[1])
    # print (len(s))
    len_s = len(s)
    isPal = 1
    for i in range(len_s):
      j=len_s-1-i
      if(i>j):
        break
      if(s[i]!=s[j]):
        isPal=0
        break
    if(isPal==1):
      print("YES ",end="")
      m = len_s%2#-(len//2)#divmod(len,2)
      if(m==0):
        print("EVEN")
      else:
        print("ODD")
    else:
      print("NO")



  exit()
